### DISCUSSION TITLE
clean-after-build flag for manifest builds

### DISCUSSION THREAD

**Discussion by greenbrettmichael** [microsoft/vcpkg#30763] [Ideas]
The clean-after-build build flag for vcpkg install is a flag that deletes various folders after a port is installed that is no longer needed. It doesn't appear that it is possible to do this while using vcpkg in manifest mode. This could reduce disk space usage on our teams and most critically, prevent CI builds from failing due to hard disk space while building the first cache after updating vcpkg. (Github Actions does not guarantee more than 20GB of hard disk space on runners, which is easily eaten by vcpkg buildtrees). 

I was not able to find what could be a good mechanism in https://github.com/microsoft/vcpkg-tool to mimic the clean-after-build behavior in manifest mode.

**Comment by autoantwort** [comment]
It also works in manifest mode. 

`vcpkg install --clean-after-build` works without problems.

**Reply by greenbrettmichael** [reply]
Yes that works, and so does manually deleting the folders in a script! However that doesn't do a very important thing, which is preserving space in process.
If you have a manifest that invokes 200 dependencies, your folder size could bloat while the vcpkg toolchain is installing those 200 dependencies. Deleting the folders or running clean-after-build after vcpkg toolchain was done wouldn't help with that problem.

**Reply by autoantwort** [reply]
https://learn.microsoft.com/en-us/vcpkg/commands/install#--clean-after-build
If cleans the folders after building a dependency and not at end end after all dependencies are build.

**Reply by autoantwort** [reply]
What do you want more?

**Reply by greenbrettmichael** [reply]
oh I see, so you would call this command before invoking the toolchain. Then those would be installed to the local vcpkg installed. I will close, thanks

**Reply by autoantwort** [reply]
> invoking the toolchain? 

Cmake integration? Then you would set [`VCPKG_INSTALL_OPTIONS`](https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_install_options) to `--clean-after-build`. For example `-DVCPKG_INSTALL_OPTIONS=--clean-after-build` or `set(VCPKG_INSTALL_OPTIONS "--clean-after-build")`

**Reply by greenbrettmichael** [reply]
exactly what I was looking for! thanks
